### ISSUE TITLE
How to set weight decay for different params

### ISSUE THREAD

**Issue by iceload123** [open-mmlab/mmdetection#105]
I want to set different weight decay for specific params. For example, when training backbone with GroupNorm， the norm params's weight decay shall be set to 0 according to paper.
How could I achieve this in config file?

**Comment by sebgao** [comment]
In config files, you cannot do it.

But you can modify `Runner` instantiation in function `_dist_train` or `_non_dist_train` of `mmdet/apis/train.py` like this:

```
non_decay_params = []
decay_params = []
for name, params in model.named_parameters():
    if 'gn' in name: #this means you must define parameters with explicit names containing 'gn'
        print('[non decay]', name)
        non_decay_params += [params]
    else:
        decay_params += [params]

import torch.optim as optim
optimizer = optim.SGD([
    {'params': non_decay_params},
    {'params': decay_params, 'weight_decay': cfg.optimizer.weight_decay},
], lr=cfg.optimizer.lr, momentum=cfg.optimizer.momentum)
runner = Runner(model, batch_processor, optimizer, cfg.work_dir,
            cfg.log_level)
```

Note that through experiments I found that these modifies do not improve at all. Correct me if wrong.
